Move @polkadot/dev to dev dependencies to support Node 18

Nutomic opened the pull request:
**Summary of changes**
Changes introduced in this pull request:
- Moves `@polkadot/dev` package to dev dependencies, so that relayer tests can be built with Node 18. Im not sure how to test if this is actually enough, or if more changes are needed.

I also tried to upgrade this package and make it build with Node 18 itself, but without success. Problem is that the latest version of polkadot/dev removes babel, and Im not sure how to replace that.

**Reference issue to close (if applicable)**
<!-- Include the issue reference this pull request is connected to -->
<!--(e.g. Closes #1)-->
References https://github.com/webb-tools/relayer/issues/456

shekohex commented:
You will probably need to update the typegen too packages too similar to https://github.com/webb-tools/protocol-substrate/pull/319

Nutomic commented:
Okay based on that example I updated the package.json, removed everything related to babel, and ran typegen (which didnt generate any changes). Hope this works.
